File: pkg/pwrgrep/testdata/fixtures/java/ExternalInput.java
import java.nio.file.Files;
import java.nio.file.Path;

public class ExternalInput {
    // ruleid: java-external-input
    public String find(@RequestParam String name) {
        return name;
    }

    void f(HttpServletRequest request) throws Exception {
        // ruleid: java-external-input
        String home = System.getenv("HOME");
        // ruleid: java-external-input
        String config = Files.readString(Path.of("config.json"));
        // ruleid: java-external-input
        String id = request.getParameter("id");
        // ok: java-external-input
        String local = "constant";
    }
}

File: pkg/pwrgrep/testdata/fixtures/csharp/ExternalInput.cs
public class ExternalInput : ControllerBase
{
    // ruleid: cs-external-input
    public IActionResult Find([FromQuery] string name) { return Ok(name); }

    void F()
    {
        // ruleid: cs-external-input
        var home = Environment.GetEnvironmentVariable("HOME");
        // ruleid: cs-external-input
        var line = Console.ReadLine();
        // ruleid: cs-external-input
        var config = File.ReadAllText("config.json");
        // ruleid: cs-external-input
        var q = Request.Query["q"];
        // ok: cs-external-input
        var local = "constant";
    }
}
